Description: Show wired connection icon if NM has an active unmanaged device
 If NM has an active unmanaged device it will forcefully set the online
 state to CONNECTED_GLOBAL. In that case show a wired connection icon
 instead of an offline icon
Author: Michael Biebl <biebl@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=471191

Index: network-manager-applet/src/applet.c
===================================================================
--- network-manager-applet.orig/src/applet.c	2013-02-09 20:46:04.969729232 +0100
+++ network-manager-applet/src/applet.c	2013-02-09 20:46:04.961729134 +0100
@@ -2745,6 +2745,10 @@
 		break;
 	default:
 		pixbuf = applet_get_device_icon_for_state (applet, &dev_tip);
+		if (!pixbuf && state == NM_STATE_CONNECTED) {
+			pixbuf = nma_icon_check_and_load ("nm-device-wired", &applet->ethernet_icon, applet);
+			g_object_ref (pixbuf);
+		}
 		break;
 	}
 
